Test failure looks related:
```
ERROR: test_bootstrap_can_be_called (corehq.apps.userreports.tests.test_pillow.StaticKafkaIndicatorPillowTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/vendor/lib/python2.7/site-packages/mock.py", line 1224, in patched
    return func(*args, **keywargs)
  File "/mnt/commcare-hq-ro/corehq/apps/userreports/tests/test_pillow.py", line 186, in test_bootstrap_can_be_called
    get_kafka_ucr_static_pillow().bootstrap()
  File "/mnt/commcare-hq-ro/corehq/apps/userreports/pillow.py", line 161, in bootstrap
    self._processor.bootstrap(configs)
  File "/mnt/commcare-hq-ro/corehq/apps/userreports/pillow.py", line 51, in bootstrap
    self.rebuild_tables_if_necessary()
  File "/mnt/commcare-hq-ro/corehq/apps/userreports/pillow.py", line 98, in rebuild_tables_if_necessary
    self.rebuild_table(sql_adapter)
  File "/mnt/commcare-hq-ro/corehq/apps/userreports/pillow.py", line 108, in rebuild_table
    rebuild_indicators.delay(sql_adapter.config.get_id)
  File "/vendor/lib/python2.7/site-packages/celery/app/task.py", line 453, in delay
    return self.apply_async(args, kwargs)
  File "/vendor/lib/python2.7/site-packages/celery/app/task.py", line 551, in apply_async
    link=link, link_error=link_error, **options)
  File "/vendor/lib/python2.7/site-packages/celery/app/task.py", line 743, in apply
    request=request, propagate=throw)
  File "/vendor/lib/python2.7/site-packages/celery/app/trace.py", line 355, in eager_trace_task
    uuid, args, kwargs, request)
  File "/vendor/lib/python2.7/site-packages/celery/app/trace.py", line 253, in trace_task
    I, R, state, retval = on_error(task_request, exc, uuid)
  File "/vendor/lib/python2.7/site-packages/celery/app/trace.py", line 240, in trace_task
    R = retval = fun(*args, **kwargs)
  File "/mnt/commcare-hq-ro/corehq/apps/userreports/tasks.py", line 38, in rebuild_indicators
    config = _get_config_by_id(indicator_config_id)
  File "/mnt/commcare-hq-ro/corehq/apps/userreports/tasks.py", line 17, in _get_config_by_id
    return StaticDataSourceConfiguration.by_id(indicator_config_id)
  File "/mnt/commcare-hq-ro/corehq/apps/userreports/models.py", line 517, in by_id
    return cls._get_from_metadata(metadata)
  File "/mnt/commcare-hq-ro/corehq/apps/userreports/models.py", line 523, in _get_from_metadata
    return cls._get_datasource_config(wrapped, metadata.domain)
  File "/mnt/commcare-hq-ro/corehq/apps/userreports/models.py", line 529, in _get_datasource_config
    doc['_id'] = cls.get_doc_id(domain, doc['table_id'])
KeyError: 'table_id'
```

--- a/corehq/apps/callcenter/data_source.py
+++ b/corehq/apps/callcenter/data_source.py
@@ -25,16 +25,18 @@ def call_center_data_source_configuration_provider():
     domains = [domain.name for domain in get_call_center_domains() if domain.use_fixtures]
     for data_source_path in data_source_paths:
         data_source_json = _get_json(data_source_path)
-        yield StaticDataSourceConfiguration(domains=domains, config=data_source_json), data_source_path
+        ds_conf = StaticDataSourceConfiguration.wrap(deepcopy(data_source_json))
+        ds_conf.domains = domains
+        yield ds_conf, data_source_path
 
 
 def get_data_source_templates():
-    call_center_data_sources = [
+    configs_json = [
         _get_json(FORM_DATA_SOURCE_PATH),
         _get_json(CASE_DATA_SOURCE_PATH),
         _get_json(CASE_ACTION_DATA_SOURCE_PATH),
     ]
-    return call_center_data_sources
+    return [config['config'] for config in configs_json]
 
 
 def get_sql_adapters_for_domain(domain_name):

--- a/corehq/apps/callcenter/data_sources/call_center_case_actions.json
+++ b/corehq/apps/callcenter/data_sources/call_center_case_actions.json
@@ -1,96 +1,99 @@
 {
-    "_id": "",
-    "domain": "",
-    "description": "",
-    "display_name": "Call center case actions",
-    "engine_id": "ucr",
-    "is_deactivated": false,
-    "named_expressions": {
-        "owner_id": {
-            "expression": {
-                "expression_if_false": {
-                    "datatype": "string",
-                    "property_name": "owner_id",
-                    "type": "property_name"
-                },
-                "expression_if_true": {
-                    "datatype": "string",
-                    "property_name": "user_id",
-                    "type": "property_name"
-                },
-                "test": {
-                    "expression": {
+    "domains": [],
+    "config": {
+        "_id": "",
+        "domain": "",
+        "description": "",
+        "display_name": "Call center case actions",
+        "engine_id": "ucr",
+        "is_deactivated": false,
+        "named_expressions": {
+            "owner_id": {
+                "expression": {
+                    "expression_if_false": {
                         "datatype": "string",
                         "property_name": "owner_id",
                         "type": "property_name"
                     },
-                    "operator": "eq",
-                    "property_value": "",
-                    "type": "boolean_expression"
-                },
-                "type": "conditional"
-            },
-            "type": "root_doc"
-        }
-    },
-    "named_filters": {},
-    "referenced_doc_type": "CommCareCase",
-    "table_id": "cc_case_actions",
-    "base_item_expression": {
-        "datatype": null,
-        "property_name": "actions",
-        "type": "property_name"
-    },
-    "configured_filter": {},
-    "configured_indicators": [
-        {
-            "column_id": "type",
-            "datatype": "string",
-            "display_name": "Case Type",
-            "expression": {
-                "expression": {
-                    "datatype": null,
-                    "property_name": "type",
-                    "type": "property_name"
+                    "expression_if_true": {
+                        "datatype": "string",
+                        "property_name": "user_id",
+                        "type": "property_name"
+                    },
+                    "test": {
+                        "expression": {
+                            "datatype": "string",
+                            "property_name": "owner_id",
+                            "type": "property_name"
+                        },
+                        "operator": "eq",
+                        "property_value": "",
+                        "type": "boolean_expression"
+                    },
+                    "type": "conditional"
                 },
                 "type": "root_doc"
-            },
-            "is_nullable": false,
-            "is_primary_key": false,
-            "transform": {},
-            "type": "expression"
+            }
         },
-        {
-            "column_id": "owner_id",
-            "datatype": "string",
-            "display_name": "Case Owner",
-            "expression": {
-                "name": "owner_id",
-                "type": "named"
-            },
-            "is_nullable": false,
-            "is_primary_key": false,
-            "transform": {},
-            "type": "expression"
+        "named_filters": {},
+        "referenced_doc_type": "CommCareCase",
+        "table_id": "cc_case_actions",
+        "base_item_expression": {
+            "datatype": null,
+            "property_name": "actions",
+            "type": "property_name"
         },
-        {
-            "column_id": "date",
-            "datatype": "datetime",
-            "display_name": "Action date",
-            "expression": {
+        "configured_filter": {},
+        "configured_indicators": [
+            {
+                "column_id": "type",
+                "datatype": "string",
+                "display_name": "Case Type",
+                "expression": {
+                    "expression": {
+                        "datatype": null,
+                        "property_name": "type",
+                        "type": "property_name"
+                    },
+                    "type": "root_doc"
+                },
+                "is_nullable": false,
+                "is_primary_key": false,
+                "transform": {},
+                "type": "expression"
+            },
+            {
+                "column_id": "owner_id",
+                "datatype": "string",
+                "display_name": "Case Owner",
+                "expression": {
+                    "name": "owner_id",
+                    "type": "named"
+                },
+                "is_nullable": false,
+                "is_primary_key": false,
+                "transform": {},
+                "type": "expression"
+            },
+            {
+                "column_id": "date",
                 "datatype": "datetime",
-                "property_name": "date",
-                "type": "property_name"
+                "display_name": "Action date",
+                "expression": {
+                    "datatype": "datetime",
+                    "property_name": "date",
+                    "type": "property_name"
+                },
+                "is_nullable": false,
+                "is_primary_key": false,
+                "transform": {},
+                "type": "expression"
             },
-            "is_nullable": false,
-            "is_primary_key": false,
-            "transform": {},
-            "type": "expression"
-        },
-        {
-            "column_id": "count",
-            "display_name": "Count",
-            "type": "count"
-        }
-    ]
+            {
+                "column_id": "count",
+                "display_name": "Count",
+                "type": "count"
+            }
+        ]
+    }
 }

--- a/corehq/apps/callcenter/data_sources/call_center_cases.json
+++ b/corehq/apps/callcenter/data_sources/call_center_cases.json
@@ -1,148 +1,151 @@
 {
-    "_id": "",
-    "domain": "",
-    "description": "",
-    "display_name": "Call center cases",
-    "engine_id": "ucr",
-    "is_deactivated": false,
-    "named_expressions": {
-        "owner_id": {
-            "expression_if_false": {
-                "datatype": "string",
-                "property_name": "owner_id",
-                "type": "property_name"
-            },
-            "expression_if_true": {
-                "datatype": "string",
-                "property_name": "user_id",
-                "type": "property_name"
-            },
-            "test": {
-                "expression": {
+    "domains": [],
+    "config": {
+        "_id": "",
+        "domain": "",
+        "description": "",
+        "display_name": "Call center cases",
+        "engine_id": "ucr",
+        "is_deactivated": false,
+        "named_expressions": {
+            "owner_id": {
+                "expression_if_false": {
                     "datatype": "string",
                     "property_name": "owner_id",
                     "type": "property_name"
                 },
-                "operator": "eq",
-                "property_value": "",
-                "type": "boolean_expression"
-            },
-            "type": "conditional"
-        }
-    },
-    "named_filters": {},
-    "referenced_doc_type": "CommCareCase",
-    "table_id": "cc_cases",
-    "base_item_expression": {},
-    "configured_filter": {},
-    "configured_indicators": [
-        {
-            "column_id": "closed",
-            "display_name": "Closed?",
-            "filter": {
-                "property_name": "closed",
-                "property_path": [],
-                "property_value": true,
-                "type": "property_match"
-            },
-            "type": "boolean"
+                "expression_if_true": {
+                    "datatype": "string",
+                    "property_name": "user_id",
+                    "type": "property_name"
+                },
+                "test": {
+                    "expression": {
+                        "datatype": "string",
+                        "property_name": "owner_id",
+                        "type": "property_name"
+                    },
+                    "operator": "eq",
+                    "property_value": "",
+                    "type": "boolean_expression"
+                },
+                "type": "conditional"
+            }
         },
-        {
-            "column_id": "owner_id",
-            "datatype": "string",
-            "display_name": "Owner ID",
-            "expression": {
-                "name": "owner_id",
-                "type": "named"
+        "named_filters": {},
+        "referenced_doc_type": "CommCareCase",
+        "table_id": "cc_cases",
+        "base_item_expression": {},
+        "configured_filter": {},
+        "configured_indicators": [
+            {
+                "column_id": "closed",
+                "display_name": "Closed?",
+                "filter": {
+                    "property_name": "closed",
+                    "property_path": [],
+                    "property_value": true,
+                    "type": "property_match"
+                },
+                "type": "boolean"
             },
-            "is_nullable": false,
-            "is_primary_key": false,
-            "transform": {},
-            "type": "expression"
-        },
-        {
-            "column_id": "type",
-            "datatype": "string",
-            "display_name": "Case Type",
-            "expression": {
-                "datatype": null,
-                "property_name": "type",
-                "type": "property_name"
+            {
+                "column_id": "owner_id",
+                "datatype": "string",
+                "display_name": "Owner ID",
+                "expression": {
+                    "name": "owner_id",
+                    "type": "named"
+                },
+                "is_nullable": false,
+                "is_primary_key": false,
+                "transform": {},
+                "type": "expression"
             },
-            "is_nullable": false,
-            "is_primary_key": false,
-            "transform": {},
-            "type": "expression"
-        },
-        {
-            "column_id": "opened_on",
-            "datatype": "date",
-            "display_name": "Date Opened",
-            "expression": {
-                "datatype": null,
-                "property_name": "opened_on",
-                "type": "property_name"
+            {
+                "column_id": "type",
+                "datatype": "string",
+                "display_name": "Case Type",
+                "expression": {
+                    "datatype": null,
+                    "property_name": "type",
+                    "type": "property_name"
+                },
+                "is_nullable": false,
+                "is_primary_key": false,
+                "transform": {},
+                "type": "expression"
             },
-            "is_nullable": false,
-            "is_primary_key": false,
-            "transform": {},
-            "type": "expression"
-        },
-        {
-            "column_id": "opened_by",
-            "datatype": "string",
-            "display_name": "Opened by",
-            "expression": {
-                "datatype": null,
-                "property_name": "opened_by",
-                "type": "property_name"
+            {
+                "column_id": "opened_on",
+                "datatype": "date",
+                "display_name": "Date Opened",
+                "expression": {
+                    "datatype": null,
+                    "property_name": "opened_on",
+                    "type": "property_name"
+                },
+                "is_nullable": false,
+                "is_primary_key": false,
+                "transform": {},
+                "type": "expression"
             },
-            "is_nullable": false,
-            "is_primary_key": false,
-            "transform": {},
-            "type": "expression"
-        },
-        {
-            "column_id": "closed_on",
-            "datatype": "date",
-            "display_name": "Date Closed",
-            "expression": {
-                "datatype": null,
-                "property_name": "closed_on",
-                "type": "property_name"
+            {
+                "column_id": "opened_by",
+                "datatype": "string",
+                "display_name": "Opened by",
+                "expression": {
+                    "datatype": null,
+                    "property_name": "opened_by",
+                    "type": "property_name"
+                },
+                "is_nullable": false,
+                "is_primary_key": false,
+                "transform": {},
+                "type": "expression"
             },
-            "is_nullable": true,
-            "is_primary_key": false,
-            "transform": {},
-            "type": "expression"
-        },
-        {
-            "column_id": "closed_by",
-            "datatype": "string",
-            "display_name": "Closed by",
-            "expression": {
-                "datatype": null,
-                "property_name": "closed_by",
-                "type": "property_name"
+            {
+                "column_id": "closed_on",
+                "datatype": "date",
+                "display_name": "Date Closed",
+                "expression": {
+                    "datatype": null,
+                    "property_name": "closed_on",
+                    "type": "property_name"
+                },
+                "is_nullable": true,
+                "is_primary_key": false,
+                "transform": {},
+                "type": "expression"
             },
-            "is_nullable": true,
-            "is_primary_key": false,
-            "transform": {},
-            "type": "expression"
-        },
-        {
-            "column_id": "server_modified_on",
-            "datatype": "date",
-            "display_name": "Server Modified",
-            "expression": {
-                "datatype": null,
-                "property_name": "server_modified_on",
-                "type": "property_name"
+            {
+                "column_id": "closed_by",
+                "datatype": "string",
+                "display_name": "Closed by",
+                "expression": {
+                    "datatype": null,
+                    "property_name": "closed_by",
+                    "type": "property_name"
+                },
+                "is_nullable": true,
+                "is_primary_key": false,
+                "transform": {},
+                "type": "expression"
             },
-            "is_nullable": true,
-            "is_primary_key": false,
-            "transform": {},
-            "type": "expression"
-        }
-    ]
+            {
+                "column_id": "server_modified_on",
+                "datatype": "date",
+                "display_name": "Server Modified",
+                "expression": {
+                    "datatype": null,
+                    "property_name": "server_modified_on",
+                    "type": "property_name"
+                },
+                "is_nullable": true,
+                "is_primary_key": false,
+                "transform": {},
+                "type": "expression"
+            }
+        ]
+    }
 }

--- a/corehq/apps/callcenter/data_sources/call_center_forms.json
+++ b/corehq/apps/callcenter/data_sources/call_center_forms.json
@@ -1,134 +1,137 @@
 {
-    "_id": "",
-    "domain": "",
-    "description": "",
-    "display_name": "Call center forms",
-    "engine_id": "ucr",
-    "named_filters": {},
-    "referenced_doc_type": "XFormInstance",
-    "table_id": "cc_forms",
-    "base_item_expression": {},
-    "configured_filter": {},
-    "named_expressions": {
-        "time_start": {
-            "datatype": "datetime",
-            "type": "property_path",
-            "property_path": [
-                "form",
-                "meta",
-                "timeStart"
-            ]
-        },
-        "time_end": {
-            "datatype": "datetime",
-            "type": "property_path",
-            "property_path": [
-                "form",
-                "meta",
-                "timeEnd"
-            ]
-        }
-    },
-    "configured_indicators": [
-        {
-            "column_id": "user_id",
-            "datatype": "string",
-            "display_name": "User ID",
-            "expression": {
-                "datatype": null,
+    "domains": [],
+    "config": {
+        "_id": "",
+        "domain": "",
+        "description": "",
+        "display_name": "Call center forms",
+        "engine_id": "ucr",
+        "named_filters": {},
+        "referenced_doc_type": "XFormInstance",
+        "table_id": "cc_forms",
+        "base_item_expression": {},
+        "configured_filter": {},
+        "named_expressions": {
+            "time_start": {
+                "datatype": "datetime",
+                "type": "property_path",
                 "property_path": [
                     "form",
                     "meta",
-                    "userID"
-                ],
-                "type": "property_path"
+                    "timeStart"
+                ]
             },
-            "is_nullable": false,
-            "is_primary_key": false,
-            "transform": {},
-            "type": "expression"
-        },
-        {
-            "column_id": "username",
-            "datatype": "string",
-            "display_name": "Username",
-            "expression": {
-                "datatype": null,
+            "time_end": {
+                "datatype": "datetime",
+                "type": "property_path",
                 "property_path": [
                     "form",
                     "meta",
-                    "username"
-                ],
-                "type": "property_path"
-            },
-            "is_nullable": true,
-            "is_primary_key": false,
-            "transform": {},
-            "type": "expression"
+                    "timeEnd"
+                ]
+            }
         },
-        {
-            "column_id": "received_on",
-            "datatype": "date",
-            "display_name": "Server Date",
-            "expression": {
-                "datatype": null,
-                "property_name": "received_on",
-                "type": "property_name"
+        "configured_indicators": [
+            {
+                "column_id": "user_id",
+                "datatype": "string",
+                "display_name": "User ID",
+                "expression": {
+                    "datatype": null,
+                    "property_path": [
+                        "form",
+                        "meta",
+                        "userID"
+                    ],
+                    "type": "property_path"
+                },
+                "is_nullable": false,
+                "is_primary_key": false,
+                "transform": {},
+                "type": "expression"
             },
-            "is_nullable": false,
-            "is_primary_key": false,
-            "transform": {},
-            "type": "expression"
-        },
-        {
-            "column_id": "time_end",
-            "datatype": "date",
-            "display_name": "Completed On",
-            "expression": {
-                "name": "time_end",
-                "type": "named"
+            {
+                "column_id": "username",
+                "datatype": "string",
+                "display_name": "Username",
+                "expression": {
+                    "datatype": null,
+                    "property_path": [
+                        "form",
+                        "meta",
+                        "username"
+                    ],
+                    "type": "property_path"
+                },
+                "is_nullable": true,
+                "is_primary_key": false,
+                "transform": {},
+                "type": "expression"
             },
-            "is_nullable": false,
-            "is_primary_key": false,
-            "transform": {},
-            "type": "expression"
-        },
-        {
-            "column_id": "duration",
-            "datatype": "integer",
-            "display_name": "Duration",
-            "expression": {
-                "type": "evaluator",
-                "statement": "timedelta_to_seconds(time_end - time_start)",
-                "context_variables": {
-                    "time_start": {
-                        "name": "time_start",
-                        "type": "named"
-                    },
-                    "time_end": {
-                        "name": "time_end",
-                        "type": "named"
+            {
+                "column_id": "received_on",
+                "datatype": "date",
+                "display_name": "Server Date",
+                "expression": {
+                    "datatype": null,
+                    "property_name": "received_on",
+                    "type": "property_name"
+                },
+                "is_nullable": false,
+                "is_primary_key": false,
+                "transform": {},
+                "type": "expression"
+            },
+            {
+                "column_id": "time_end",
+                "datatype": "date",
+                "display_name": "Completed On",
+                "expression": {
+                    "name": "time_end",
+                    "type": "named"
+                },
+                "is_nullable": false,
+                "is_primary_key": false,
+                "transform": {},
+                "type": "expression"
+            },
+            {
+                "column_id": "duration",
+                "datatype": "integer",
+                "display_name": "Duration",
+                "expression": {
+                    "type": "evaluator",
+                    "statement": "timedelta_to_seconds(time_end - time_start)",
+                    "context_variables": {
+                        "time_start": {
+                            "name": "time_start",
+                            "type": "named"
+                        },
+                        "time_end": {
+                            "name": "time_end",
+                            "type": "named"
+                        }
                     }
-                }
+                },
+                "is_nullable": false,
+                "is_primary_key": false,
+                "transform": {},
+                "type": "expression"
             },
-            "is_nullable": false,
-            "is_primary_key": false,
-            "transform": {},
-            "type": "expression"
-        },
-        {
-            "column_id": "xmlns",
-            "datatype": "string",
-            "display_name": "xmlns",
-            "expression": {
-                "datatype": null,
-                "property_name": "xmlns",
-                "type": "property_name"
-            },
-            "is_nullable": false,
-            "is_primary_key": false,
-            "transform": {},
-            "type": "expression"
-        }
-    ]
+            {
+                "column_id": "xmlns",
+                "datatype": "string",
+                "display_name": "xmlns",
+                "expression": {
+                    "datatype": null,
+                    "property_name": "xmlns",
+                    "type": "property_name"
+                },
+                "is_nullable": false,
+                "is_primary_key": false,
+                "transform": {},
+                "type": "expression"
+            }
+        ]
+    }
 }

--- a/corehq/apps/userreports/models.py
+++ b/corehq/apps/userreports/models.py
@@ -486,7 +486,7 @@ class StaticDataSourceConfiguration(JsonObject):
 
     @classmethod
     def all(cls):
-        for wrapped, _ in cls._all():
+        for wrapped, path in cls._all():
             for domain in wrapped.domains:
                 yield cls._get_datasource_config(wrapped, domain)
 
@@ -566,7 +566,7 @@ class StaticReportConfiguration(JsonObject):
 
     @classmethod
     def all(cls):
-        for wrapped, _ in StaticReportConfiguration._all():
+        for wrapped, path in StaticReportConfiguration._all():
             for domain in wrapped.domains:
                 yield cls._get_report_config(wrapped, domain)
 
@@ -613,7 +613,7 @@ class StaticReportConfiguration(JsonObject):
 
     @classmethod
     def report_class_by_domain_and_id(cls, domain, config_id):
-        for wrapped, _ in cls._all():
+        for wrapped, path in cls._all():
             if cls.get_doc_id(domain, wrapped.report_id, wrapped.custom_configurable_report) == config_id:
                 return wrapped.custom_configurable_report
         raise BadSpecError(_('The report configuration referenced by this report could '

--- a/corehq/apps/userreports/tests/test_static_data_sources.py
+++ b/corehq/apps/userreports/tests/test_static_data_sources.py
@@ -29,7 +29,7 @@ class TestStaticDataSource(SimpleTestCase, TestFileMixin):
             for config in all[:2]:
                 self.assertEqual('all_candidates', config.table_id)
 
-            for config in all[3:]:
+            for config in all[2:]:
                 self.assertEqual('cc1', config.domain)
 
     def test_is_static_positive(self):

